cpp/src/arrow/compute/kernels/scalar_if_else_benchmark.cc

```diff
@@ -57,9 +57,45 @@ struct GetBytesProcessed<Type, enable_if_base_binary<Type>> {
   }
 };
 
+template <typename DataType>
+struct GetBytesProcessed<DataType, enable_if_list_like<DataType>> {
+  static int64_t Get(const std::shared_ptr<Array>& arr) {
+    using OffsetType = typename TypeTraits<DataType>::OffsetType::c_type;
+
+    int64_t total_bytes = 0;
+    if constexpr (DataType::type_id == Type::LIST ||
+                  DataType::type_id == Type::LARGE_LIST) {
+      total_bytes += (arr->length() + 1) * sizeof(OffsetType);
+    } else {
+      ADD_FAILURE() << "GetBytesProcessed dispatching not implemented for "
```

Comment:
One alternative is (you need to include `arrow/util/logging.h` if not already done):
```suggestion
      ARROW_LOG(FATAL) << "GetBytesProcessed dispatching not implemented for "
```

Reply:
Done. Confirmed this works even in Release builds.